helios-tools/src/main/java/com/spotify/helios/cli/command/PortMappingParser.java

```diff
@@ -0,0 +1,109 @@
+/*-
+ * -\-\-
+ * Helios Tools
+ * --
+ * Copyright (C) 2016 - 2017 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.helios.cli.command;
+
+import static com.google.common.base.Optional.fromNullable;
+import static com.spotify.helios.common.descriptors.PortMapping.TCP;
+import static com.spotify.helios.common.descriptors.PortMapping.WILDCARD_ADDRESS;
+import static java.util.regex.Pattern.compile;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import com.spotify.helios.common.descriptors.PortMapping;
+
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+class PortMappingParser {
+
+  private static final Pattern PATTERN =
+      compile("(?<n>[_\\-\\w]+)=((?<ip>(localhost|([0-9]{1,3}.){3}[0-9]{1,3})):)"
+              + "?(?<i>\\d+)(:(?<e>\\d+))?(/(?<p>\\w+))?");
```

Comment:
do we definitely still need a regex here? A regex that checks if something is a valid IP address feels prone to failure to me. Would it be easier (and more correct) to split the string up by various tokens and then pass the ip address portion to the guava `boolean isInetAddress(string)` function?